api/src/test/java/org/openmrs/api/PatientServiceTest.java

```diff
@@ -3258,5 +3263,53 @@ public void mergePatients_shouldMoveAllObsWithSameHierarchy() throws Exception {
 		assertEquals(1, encounterService.getEncounter(57).getObsAtTopLevel(false).size());
 		assertEquals(2, encounterService.getEncounter(57).getObsAtTopLevel(true).size());
 	}
+	
+	@Test
+	public void getDuplicatePatientsByAttributes_shouldThrowExceptionIfAttributeListIsNull() throws Exception{
+		expectedException.expect(APIException.class);
+		expectedException.expectMessage("There must be at least one attribute supplied to search on");
+		patientService.getDuplicatePatientsByAttributes(null);
+	}
+
+	@Test
+	public void getDuplicatePatientsByAttributes_shouldThrowExceptionIfAttributeListIsEmpty() throws Exception{
+		expectedException.expect(APIException.class);
+		expectedException.expectMessage("There must be at least one attribute supplied to search on");
+		List<String> attributes = new ArrayList<>();
+		patientService.getDuplicatePatientsByAttributes(attributes);
+	}
+	
+	@Test
+	public void processDeath_shouldThrowExceptionIfPatientIsNull() throws Exception {
+		Date dateOfDeath = new Date();
+		Concept causeOfDeath = Context.getConceptService().getConcept(1);
```

Comment:
Can you mock this service ? https://wiki.openmrs.org/display/docs/Mock+Doc

Reply:
I cant mock these tests because even if I do, the context is called from multiple other tests in this class. That would mean I would have to edit all of them as well and I don't think thats a good idea.